util/unifyfs/src/unifyfs-rm.c

```diff
@@ -539,6 +657,45 @@ static size_t construct_server_argv(unifyfs_args_t* args,
     return argc;
 }
 
+/**
+ * @brief Default data stage launch routine
+ *
+ * @param args        The command-line options
+ * @param stage_args  unifyfs-stage argument vector to be filled
+ *
+ * @return number of server arguments
+ */
+static size_t construct_stage_argv(unifyfs_args_t* args,
+                                   char** stage_argv)
+{
+    size_t argc = 0;
+
+    if (stage_argv != NULL) {
+        stage_argv[0] = strdup(LIBEXECDIR "/unifyfs-stage");
+    }
+    argc = 1;
+
+    if (args->mountpoint != NULL) {
+        if (stage_argv != NULL) {
+            stage_argv[argc] = strdup("-m");
+            stage_argv[argc + 1] = strdup(args->mountpoint);
+        }
+        argc += 2;
```

Comment:
If `args->mountpoint != NULL && stage_argv == NULL`, should we still be doing `argc += 2`?

Reply:
That's a good question.   I don't really know.  But again, that mirrors the structure of construct_server_argv, which is constructed similarly.  

I have a counter question to this.  The last block in both construct_server_argv and construct_stage_argv has the argc+= stage OUTSIDE the whole block.  I can't think that's actually correct or what was intended.  Here's the one at the bottom of (existing function) construct_server_argv:
    if (server_argv != NULL) {
        server_argv[argc] = strdup("-S");
        server_argv[argc + 1] = strdup(args->share_dir);
        server_argv[argc + 2] = strdup("-H");
        server_argv[argc + 3] = strdup(args->share_hostfile);
    }
    argc += 4;

Reply:
@craigsteffen @tonyhutter `construct_server_argv()` is used in two modes. The first mode is when `server_argv == NULL`, in which it just returns the count of arguments that will reside in `server_argv`. That is why `argc` is incremented outside the checks for NULL. The returned argc count is then used to allocate the actual server arguments array, and the function is called again in the second mode, which fills in the argv array. The reason the return value is `size_t` is because that is the appropriate type for the size of an array (and also the size parameter to malloc/calloc).

Reply:
I added a comment to the source indicating that.  Thanks Mike.